### PR TITLE
User controller unit test added, 100% coverage

### PR DESCRIPTION
**Pull request by EllyChanx**
User controller test coverage 100% with unit tests

### CODE REVIEW

**Review by lucafrancesc — APPROVED** [review]
User controller tested

**Review by mateuszstacel — APPROVED** [review]
Well Done ! Elly :)

File: spec/controllers/posts_controller_spec.rb [hunk 1]
```diff
@@ -4,7 +4,6 @@
 RSpec.describe PostsController, type: :controller do
 
   it 'should show post' do
-    post = Post.new
```
Comment (EdwardAndress) [inline spec/controllers/posts_controller_spec.rb:7]: 👍

File: spec/controllers/users_controller_spec.rb [hunk 2]
```diff
@@ -2,11 +2,43 @@
...
     end
   end
 
+  describe "GET #show" do
+    it "assign user return as @user" do
+      get :show, params: { id: @user.id }
+    end
+  end
+
+  describe "GET #edit" do
```
Comment (EdwardAndress) [inline spec/controllers/users_controller_spec.rb:25]: It should also be possible to test that the controller renders the correct template

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -2,11 +2,43 @@
 
 RSpec.describe UsersController, type: :controller do
 
+  before(:each) do
```
Comment (EdwardAndress) [inline spec/controllers/users_controller_spec.rb:5]: Nice!

File: spec/controllers/users_controller_spec.rb [hunk 4]
```diff
@@ -2,11 +2,43 @@
...
+
   describe "GET #index" do
     it "returns http success" do
       get :index
       expect(response).to have_http_status(:success)
     end
   end
 
+  describe "GET #show" do
+    it "assign user return as @user" do
```
Comment (EdwardAndress) [inline spec/controllers/users_controller_spec.rb:20]: There's no expectation here.

File: spec/controllers/users_controller_spec.rb [hunk 5]
```diff
@@ -2,11 +2,43 @@
...
+    end
+  end
+
+  describe "PUT update" do
+    it "updates the requested user" do
+      put :update, :params => {:id => @user.id, :user => @user.attributes = { :email => "new@new.com", :password => "1234567" }}
+      expect(@user.email).to eq("new@new.com") 
+      expect(response).to redirect_to root_url
+      # expect(response).to redirect_to ('/')
+    end
```
Comment (EdwardAndress) [inline spec/controllers/users_controller_spec.rb:38]: how about something like this?

```
let(:update_params) { <write the correct params in here> }

it "updates the requested user" do
  expect { put :update, params: update_params }.to change { @user.email }.from("el@el.com").to("new@new.com")
end
```
Moving the params to a let block allows you to use this one liner without the line becoming too long.  The one liner ensures the correct transition is made; from the old state to the new state.

**Pull request merged**